docs/BUILD.bazel

```diff
@@ -112,7 +134,7 @@ genrule(
     name = "flatten_header_vm",
     outs = ["flatten_header.vm"],
     cmd = ">$@ echo '# Rust rules\n\n%s\n'" % "\n".join(
-        sorted(["* [{rule}](#{rule})".format(rule = v) for k in PAGES.keys() for v in PAGES[k]]),
+        sorted(["- [{rule}](#{rule})".format(rule = v) for k in PAGES.keys() for v in PAGES[k].symbols]),
```

Comment:
why change `*` to `-` when it renders the same way?

Reply:
It's the default behavior of https://github.com/DavidAnson/markdownlint which seems to have some good justifications for each rule it has so I thought it'd be an improvement.

Reply:
Checked a couple other places throughout the repository, as well as my teams, and `*` is used pretty much exclusively.

Looking at [M0004](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md004), that repository's default "consistent" not any particular item (though asterisk is listed next).

For familiarity of folks here and reducing churn, would you mind going with `*`?